ignite/metrics/gan/fid.py

```diff
@@ -0,0 +1,177 @@
+from typing import Callable, Sequence, Union
+
+import numpy as np
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["FID", "InceptionExtractor"]
+
+
+def fid_score(
+    mu1: torch.Tensor, mu2: torch.Tensor, sigma1: torch.Tensor, sigma2: torch.Tensor, eps: float = 1e-6
+) -> float:
+
+    diff = mu1 - mu2
+    import scipy
+
+    # Product might be almost singular
+    covmean, _ = scipy.linalg.sqrtm(sigma1.mm(sigma2), disp=False)
+    if not np.isfinite(covmean).all():
+        offset = np.eye(sigma1.shape[0]) * eps
+        covmean = scipy.linalg.sqrtm((sigma1 + offset).dot(sigma2 + offset))
+
+    # Numerical error might give slight imaginary component
+    if np.iscomplexobj(covmean):
+        if not np.allclose(np.diagonal(covmean).imag, 0, atol=1e-3):
+            m = np.max(np.abs(covmean.imag))
+            raise ValueError("Imaginary component {}".format(m))
+        covmean = covmean.real
+
+    tr_covmean = np.trace(covmean)
+
+    return diff.dot(diff) + np.trace(sigma1) + np.trace(sigma2) - 2 * tr_covmean
+
+
+class InceptionExtractor:
+    def __init__(self) -> None:
+        from torchvision import models
+
+        self.model = models.inception_v3(init_weights=True)
+        self.model.fc = torch.nn.Sequential()
+        self.model.eval()
+
+    def __call__(self, data: torch.Tensor) -> torch.Tensor:
+        if data.shape[1:] != (3, 299, 299):
```

Comment:
If inception model is headless (no fully-connected layer), can't we use any size input. In the metric code we just have to ensure that we always have the same size...

Reply:
299x299 is minimum size of image needed I can change it to >= 299 if needed

Reply:
> If inception model is headless (no fully-connected layer), can't we use any size input. 

That's right ! 

> In the metric code we just have to ensure that we always have the same size...

The user gives the feature size, so there is a test on shapes during iteration.